### PR TITLE
Include Harfbuzz sources in source distribution

### LINKED ISSUE #61: 0.13.1 source distribution fails to build, missing header files
Using the newly published 0.13.1 source distribution from PyPi:

```console
$ python setup.py build
running build
running build_py
copying src/uharfbuzz/_version.py -> build/lib.linux-x86_64-3.8/uharfbuzz
running build_ext
building 'uharfbuzz._harfbuzz' extension
gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fno-semantic-interposition -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -D_FORTIFY_SOURCE=2 -fPIC -DHB_NO_MT=1 -DHB_EXPERIMENTAL_API=1 -Iharfbuzz/src -I/usr/include/python3.8 -c src/uharfbuzz/_harfbuzz.cpp -o build/temp.linux-x86_64-3.8/src/uharfbuzz/_harfbuzz.o -std=c++11
src/uharfbuzz/_harfbuzz.cpp:654:10: fatal error: hb.h: No such file or directory
  654 | #include "hb.h"
      |          ^~~~~~
compilation terminated.
error: command 'gcc' failed with exit status 1
```

The extracted package doesn't seem to have the referenced `hb.h` header file at all. In fact it doesn't include any of the sources needed to build Harfbuzz.

### PR DESCRIPTION
**Pull request by alerque**
Closes #61.

Relevant Python [docs here](https://packaging.python.org/guides/using-manifest-in/).

Tested locally by building a source dist (`python setup.py sdist`), then extracting the zip to a new folder and building (`python setup.py build`).

### CODE REVIEW

**Comment by justvanrossum** [comment]
Thanks for the quick fix! I will just merge this and release v0.13.2.

**Comment by anthrotype** [comment]
uharfbuzz already uses setuptools_scm to include all files under version control (except those explicitly excluded) to the sdist package.
However, setuptools_scm does not also include files in git submodules (see https://github.com/pypa/setuptools_scm/issues/206).

To workaround this, some time ago I have created this setuptools extension that finds all git tracked files, including inside submodules and adds them to the sdist manifest (the command used is `git ls-files --cached --recurse-submodule`). 

https://github.com/anthrotype/setuptools_git_ls_files

To enable this for uharfbuzz you simply need to add "setuptools-git-ls-files" to both pyproject.toml build-system requires, and to setup.py setup_requires, like I did in cffsubr below

https://github.com/adobe-type-tools/cffsubr/commit/855d26c3595ca5f311df01efffc591ed79077d2d

You won't need to maintain the MANIFEST.in file any more as git will take care of tracking files.

File: MANIFEST.in [hunk 1]
```diff
@@ -0,0 +1 @@
+include harfbuzz/src/*
```
Comment (anthrotype) [inline MANIFEST.in:1]: a blind include everything under harfbuzz/src may also end up including temporary files generated by the build or similar left-overs found there by accident

**Comment by justvanrossum** [comment]
Ouch, sorry for moving too quickly and possibly messing things up. Should I revert? And what about the release I just did?

**Comment by anthrotype** [comment]
I don't think you need to revert, as long as the 0.13.1 works for you. Mine was a suggestion for an improved method for managing the sdist manifest

**Comment by justvanrossum** [comment]
Ok, I'll leave it at this then. I hope the 0.13.2 release works for @alerque.

**Comment by alerque** [comment]
You can iterate on this however you like, but the previous two releases' source distributions were completely unusable. "Go directly to jail, do not pass GO" kind of unusable. I used what the documented Python packaging method was, if you want to swap that out for a custom plugin I don't care as long as the releases are usable. 0.13.2 hasn't been posted to PyPi yet, but I'll check it as soon as it is.

By the way if you do go down that road I don't think you want to include all the harfbuzz sources because you aren't using any of the machinery in there, only a shortcut to compile the amalgam file—hence just the stuff in the src dir is needed, not the whole list of files tracked in the git submodule.

**Comment by alerque** [comment]
Also looking at the source this line stood out:

https://github.com/harfbuzz/uharfbuzz/blob/9c5c8c5f1e2752d5ba72ef0c43a7bb19c319be44/setup.py#L29

But apparently that's only used for setting include parameters of the `c++` build run, it doesn't affect files included in the dist. The `harfbuzz.cc` file mentioned on the next line:

https://github.com/harfbuzz/uharfbuzz/blob/9c5c8c5f1e2752d5ba72ef0c43a7bb19c319be44/setup.py#L30

...does get included so it's the source dist is paying some attention to what is used there, but not *enough* attention and I couldn't find any docs on how to make it follow imported sources to get a complete dependency list.

**Comment by anthrotype** [comment]
> just the stuff in the src dir is needed, not the whole list of files tracked in the git submodule

fair enough. Maybe you could use a glob pattern to only include source and header files by their file extension.

**Comment by alerque** [comment]
Release v0.13.2 as posted on PyPi works .... finally.

**Pull request merged**